app/models/concerns/node_shared.rb

```diff
@@ -53,6 +53,23 @@ def self.graph_grid(body, _page = 1)
     end
   end
 
+  def self.simple_data_grapher(body, _page = 1)
+    body.gsub(/(?<![\>`])(\<p\>)?\[simple-data-grapher\:(\S+)\]/) do |_tagname|
+      ids = Regexp.last_match(2)
+      a = ActionController::Base.new
+      randomSeed = rand(1000).to_s
+      output = a.render_to_string(template: "grids/_simple-data-grapher",
+                                  layout:   false,
+                                  locals: {
+                                    ids: ids,
+                                    randomSeed: randomSeed,
+                                    idName: 'sdg-graph-' + randomSeed,
+                                    type: "graph"
+                                  })
+      output
```

Comment:
I guess this `output` is redundant. Please confirm.

Reply:
The output is being returned here. Its a return statement like rest of the code in `node_shared.rb`